docs/graphs/graph-concepts/graphs-and-indexes.md

```diff
@@ -0,0 +1,105 @@
+
+A skiplist index is suitable for single or multiple attributes, supporting both equality lookups and range queries. It also enables sorting by the indexed attribute(s). If your graph queries involve filtering or sorting vertices or edges based on one or multiple attributes' values, particularly when range queries are involved, a skiplist index can be helpful.
+
+**Use case**: Locating posts within a specific date range or sorting edges by a weight attribute.
+
+### Persistent Index
+
+A persistent index is similar to a skiplist index in terms of supported queries, but it is stored on disk rather than in memory. This index type is useful for large datasets that do not fit in memory, but it might have slightly higher access latency compared to in-memory indexes. If your graph queries involve large datasets and require filtering or sorting based on attribute values, then a persistent index might be an appropriate choice.
+
+**Use case**: Large graphs where the index size exceeds available memory, and queries involve sorting or filtering based on attribute values.
+
+### Fulltext Index
+
+A fulltext index is designed for text-based searches, supporting word or phrase searches within the indexed attributes. If your graph queries involve searching for vertices or edges based on text content, then a fulltext index can be useful.
+
+**Use case**: Searching for posts containing specific keywords or finding users based on their interests.
+
+## Example: Using Indexes with Graphs
+
+In this theoretical example, a simple graph with two vertex collections and one edge collection is created, indexes are added to optimize query performance, and then a graph query is executed using those indexes. The graph represents a social network with users and posts, where users can create posts and follow other users.
+
+### Creating Collections
+
+Before traversing graphs, collections need to be created and data added to them:
+
+- Vertex (document) collection `users` - stores user information.
+- Vertex (document) collection `posts` - stores post information.
+- Edge collection `relations` - stores relationships between users (follows) and user-post relationships (creates).
+
+For more information about creating collections, refer to [Collections](../../collections/).
+
+### Adding Indexes
+
+After creating these collections and inserting sample data, indexes can be added to optimize query performance.
+
+For more information on adding indexes, refer to [Working with Indexes](../../collections/indexing/working-with-indexes) or the [gdnsl index](../../cli/indexes-cli) command.
+
+Add the following indexes:
+
+- **email_index**: A "hash" index on the `email` attribute in the `users` collection. This index would speed up queries that filter users based on their email addresses.
+- **from_index**: A "hash" index on the `_from` attribute in the `relations` collection. This index would optimize queries that involve traversing edges based on their starting vertices.
```

Comment:
_from is already indexed automatically with the edge index; I don't think it is necessary to add an additional index